content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+In the scoping section, we mentioned briefly that we wanted the game to spawn obstacles _from time to time_ (if you played Chrome's dinosaur game, you know they appear randomly). There's a level of uncertainty in that statement that we can resolve by getting random values for the next time for an object to appear:
+
+```ts
+    static GetScoreNextObstacle(): number
+    {
+        // let's have a minimum distance so objects don't appear next to each other
+        let minimumDistance = 25;
+        // we can define a level of difficulty to make it harder as we go on (limit is 5)
+        let difficulty = Math.min(this.Score / 100, 5);
+        // define the random value based on values above
+        return (Math.random() * 10 - (difficulty * 4)) + minimumDistance;
+    }
+```
+
+### AddObject()
+
+Since we are using this method to generalize adding decorations (i.e: non-solid objects) and obstacles, we let the caller decide which height (or Y position) it should be spawned at, but the X position should always be set as the game's width (at the right limit of the screen).
+The other thing we need to keep in mind is that if we want the object to update, we need to add it to the list of `ActiveEntities` so that the game keeps track of it (and to PixiJS's Stage object).
+
+So the method would look something like this:
+
+```ts
+    private static AddObject(spriteName: keyof Sprites, height: number, isSolid: boolean) {
+        let obstacle = new ScrollingObject(spriteName, GameApp.Width, height, isSolid);
+        GameApp.ActiveEntities.push(obstacle);
+        GameApp.Stage.addChild(obstacle.sprite);
+    }
+```
+
+## Collisions
+
+We are at the point where our game feels pretty playable, we can run it and jump over objects. A "small" problem that you might have noticed, however, is that we cannot really bump into objects and lose the game, so we'll have to find a solution for that.
+
+Our sprites are loaded as 2D textures that fit inside a box which has a specific width and height. PixiJS gives us access to these values (and the current X and Y positions) through a rectangle structure, accessible through `PIXI.Sprite.getBounds()`. Imagine we have 2 rectangles of which we know these 4 values (i.e: X position, Y position, width and height), how can we tell if they are overlapping/colliding? One way of answering is by inversion (_When do they **not** collide?_):
+
+- If the the first rectangle's X position starts after the second rectangle's X extension ends, then they are definitely not colliding because of the orthogonal properties of the X and Y axes.
+  - This also applies the other way around (if the second rectangle's X position starts after the first one's X extension ends they don't collide)
+- If the the first rectangle's Y position starts after the second rectangle's Y extension ends, the same applies.
+  - And the other way around.
+
+Since we know when they are **not** colliding, we can negate that and get the result.
```

Comment:
This is like the BEST explanation of collision detection I have ever read (and I have read a few back in the day) awesome job